tests/scripts/all.sh

```diff
@@ -469,101 +467,62 @@ tests/scripts/doxygen.sh
 
 
 ################################################################
-#### Build and test many configurations and targets
+#### 1 Basic tests
 ################################################################
 
-if [ $RUN_ARMCC -ne 0 ] && [ $YOTTA -ne 0 ]; then
-    # Note - use of yotta is deprecated, and yotta also requires armcc to be on the
-    # path, and uses whatever version of armcc it finds there.
-    msg "build: create and build yotta module" # ~ 30s
-    cleanup
-    record_status tests/scripts/yotta-build.sh
-fi
+# 1.1 Basic tests: default config
+#################################
 
-msg "build: cmake, gcc, ASan" # ~ 1 min 50s
+msg "build: cmake, gcc, ASan" # ~ 1 min
 cleanup
 CC=gcc cmake -D CMAKE_BUILD_TYPE:String=Asan .
 make
 
-msg "test: main suites (inc. selftests) (ASan build)" # ~ 50s
+msg "test: main suites (inc. selftests) (ASan build)" # < 10s
 make test
 
-msg "test: ssl-opt.sh (ASan build)" # ~ 1 min
+msg "test: ssl-opt.sh (ASan build)" # ~ 2 min
 if_build_succeeded tests/ssl-opt.sh
 
-msg "test/build: ref-configs (ASan build)" # ~ 6 min 20s
-record_status tests/scripts/test-ref-configs.pl
-
-msg "build: with ASan (rebuild after ref-configs)" # ~ 1 min
-make
```

Comment:
Why did we have this build and no longer need it?

Reply:
We had this build because we chose to run `test-ref-configs.pl` first, but this test does many builds with non-default config, so after running it we needed to re-build with the default config and previous build options before we could go on and run `compat.sh` on the default config with ASan. Now that the `test-ref-configs.pl` has been moved to another section, we only need one build for the whole `make test; ssl-opt.sh; compat.sh` sequence of tests.